### PR TITLE
Add Wallaby integration

### LINKED ISSUE #1492: [Task] Consider using Wallaby.js
Recently I stumbled across [Wallaby.js](https://wallabyjs.com/) and it looks like a good addition to our `DevOps`.

`VSCode Extension`: https://marketplace.visualstudio.com/items?itemName=WallabyJs.wallaby-vscode

@vscode-icons/development What's your opinion?

### PR DESCRIPTION
**Pull request by JimiC**
_**Resolves #1492**_

Wow. `Wallaby.js` is so much more than `Mocha` itself. Finally, the test runner I was looking for. This tool sheds light on every "dark" corner of your project. It shows you the **real** code coverage.

### CODE REVIEW

File: wallaby.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      done(file.rename(`../${file.path}`).content)
+    }
+  },
+  hints: {
+    ignoreCoverage: /\/* wallaby ignore next\/*/
+  },
+  testFramework: "mocha",
+  env: {
+    type: "node",
+    runner: `${require('os').homedir()}/.nvm/versions/node/v7.9.0/bin/node`
```
Comment (robertohuertasm) [inline wallaby.js]: I already merged this and forgot to tell you that this isn't working in Windows environments. My nvm gets installed in `userDir`\AppData\Roaming\nvm

Comment (JimiC) [reply wallaby.js]: Honestly I didn't test it in Windows as I do all development in Linux lately. I'll have a look when I find some time.

Comment (robertohuertasm) [reply wallaby.js]: I've submitted a PR removing the runner line. I think we can safely remove it. Devs using wallaby (namely you, @jens1o and me) already know which is the node version supported so we are responsible of selecting the approppriate node version in our machines.

Comment (jens1o) [reply wallaby.js]: tbh I didn't set it up yet :D

**Pull request merged**